File: app/src/main/java/com/christian/ChristianApplication.java
package com.christian;

import android.app.Application;

import com.christian.util.CrashHandler;

import org.xutils.x;

import java.lang.ref.WeakReference;

/**
 * author：Administrator on 2017/4/2 00:20
 * email：lanhuaguizha@gmail.com
 */

public class ChristianApplication extends Application {

    // 在application的onCreate中初始化
    @Override
    public void onCreate() {
        super.onCreate();
        x.Ext.init(this);
        x.Ext.setDebug(BuildConfig.DEBUG); // 是否输出debug日志, 开启debug会影响性能.
        // CrashHandler
        WeakReference<CrashHandler> crashHandler = CrashHandler.getInstance();
        crashHandler.get().init(this);
    }
}

File: app/src/main/kotlin/com/christian/ChristianApplication.kt
package com.christian

import android.app.Application
import androidx.multidex.MultiDex
import com.christian.util.CrashHandler
import com.github.anzewei.parallaxbacklayout.ParallaxHelper

class ChristianApplication : Application() {

    init {
        context = this
    }

    /**
     * Gets the application context.
     *
     * @return the application context
     */
    companion object {
        lateinit var context: ChristianApplication
    }

    override fun onCreate() {
        super.onCreate()
        // leak canary
        MultiDex.install(this)
        // Normal app init code...
        val crashHandler = CrashHandler.getInstance()
        crashHandler.get()?.init(this)
        registerActivityLifecycleCallbacks(ParallaxHelper.getInstance())
        // Dark Mode
//        SkinCompatManager.withoutActivity(this)                         // 基础控件换肤初始化
//                .addInflater(SkinMaterialViewInflater())            // material design 控件换肤初始化[可选]
//                .addInflater(SkinConstraintViewInflater())          // ConstraintLayout 控件换肤初始化[可选]
//                .addInflater(SkinCardViewInflater())                // CardView v7 控件换肤初始化[可选]
////                .setSkinStatusBarColorEnable(false)                     // 关闭状态栏换肤，默认打开[可选]
////                .setSkinWindowBackgroundEnable(false)                   // 关闭windowBackground换肤，默认打开[可选]
//                .loadSkin()
    }
}


